### PR TITLE
added missing call to onDispatched for instrumentation context of beginExecution - resolves #1620

### LINKED ISSUE #1620: InstrumentationContext onDispatched not called for parsing
**Describe the bug**
For a couple of the methods on Instrumentation (beginParse and beginValidation) the onDispatched method on the returned InstrumentationContext is not called.  

**To Reproduce**
My Instrumentation, which is later passed into a GraphQL instance:
```
public class TestInstrumentation extends SimpleInstrumentation {
  @Override
  public InstrumentationContext<Document> beginParse(InstrumentationExecutionParameters parameters) {
    return new InstrumentationContext<>() {
      @Override
      public void onDispatched(CompletableFuture<Document> result) {
        System.out.println("Dispatched");
      }

      @Override
      public void onCompleted(Document result, Throwable t) {
        System.out.println("Completed");
      }
    };
  }
}
```

Expected:
```
Dispatched
Completed
```

Actual:
```
Completed
```

### PR DESCRIPTION
**Pull request by bsara**
Added a call to `onDispatched` for the `beginExecution` instrumentation context.

resolves #1620

### CODE REVIEW

File: src/main/java/graphql/GraphQL.java [hunk 1]
```diff
@@ -519,6 +519,9 @@ public ExecutionResult execute(ExecutionInput executionInput) {
             GraphQLSchema graphQLSchema = instrumentation.instrumentSchema(this.graphQLSchema, instrumentationParameters);
 
             CompletableFuture<ExecutionResult> executionResult = parseValidateAndExecute(executionInput, graphQLSchema, instrumentationState);
+
+            executionInstrumentation.onDispatched(executionResult);
+
             //
```
[the remark this answers is not in the document]
Comment (bsara) [reply src/main/java/graphql/GraphQL.java:525]: Awesome. I'll make the change. Thank you for the quick turn around.

[the remark this answers is not in the document]
Comment (bsara) [reply src/main/java/graphql/GraphQL.java:524]: I can't seem to get the tests to run locally for me (something with the gradle daemon). Could I get approval for this to run the tests? I don't want to move this from draft status until I can successfully test it of course.

File: src/main/java/graphql/GraphQL.java [hunk 2]
```diff
@@ -513,15 +513,24 @@ public ExecutionResult execute(ExecutionInput executionInput) {
...
             // finish up instrumentation
-            executionResult = executionResult.whenComplete(executionInstrumentation::onCompleted);
+            executionResult = executionResult.whenComplete((result, throwable) -> {
+                if (throwable != null) {
+                    beginExecutionCF.completeExceptionally(throwable);
+                } else {
+                    beginExecutionCF.complete(result);
+                }
+                executionInstrumentation.onCompleted(result, throwable);
+            });
```
Comment (bbakerman) [inline src/main/java/graphql/GraphQL.java:533]: yeah that is better than what I proposed

File: src/main/java/graphql/GraphQL.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bbakerman) [reply src/main/java/graphql/GraphQL.java:524]: I tend to run the tests inside IDEA - albeit `.gradlew clean build` also works for me

I have approved the build and hence the github actions will run the builds

**Comment by bbakerman** [comment]
SubscriptionExecutionStrategyTest > instrumentation gets called on subscriptions FAILED
ChainedInstrumentationStateTest > basic chaining and state management FAILED
InstrumentationTest > Instrumentation of simple serial execution FAILED
PreparsedDocumentProviderTest > Preparse of simple serial execution FAILED

Lots of tests that assert instrument test call chain are failing.  You really need to work out your local test env to runs tests

**Comment by bbakerman** [comment]
Or just keep commiting and allow github to run them

**Comment by bsara** [comment]
I'll get it running locally.

**Comment by bbakerman** [comment]
See https://github.com/graphql-java/graphql-java/pull/2730

**Comment by bbakerman** [comment]
Thanks for the original PR - I took it and tweaked it a little and fixed the tests

**Pull request closed without merging**